sympy/matrices/matrices.py

```diff
@@ -1577,6 +1578,270 @@ def denom_clean(l):
             ret = [(val.evalf(chop=chop), mult, [v.evalf(chop=chop) for v in es]) for val, mult, es in ret]
         return ret
 
+    def eigenvects_mod_square_elementwise(self, simplify=False, dotprodsimp=None):
+        """Return list of triples (eigenval, multiplicity, eigenspace).
+
+        ``eigenspace`` contains a single ``eigenvector`` with the elements being
+        the squares of the absolute values of the elements of the actual
+        eigenvector. This only works for hermitian matrices with eigenvalues
+        of multiplicity 1.
+
+        Parameters
+        ==========
+
+        simplify : bool or function, optional
+            If it is set to ``True``, it attempts to return the most
+            simplified form of expressions returned by applying default
+            simplification method in every routine.
+
+            If it is set to ``False``, it will skip simplification in this
+            particular routine to save computation resources.
+
+            If a function is passed to, it will attempt to apply
+            the particular function as simplification method.
+
+        dotprodsimp : bool, optional
+            Specifies whether intermediate term algebraic simplification is used
+            during matrix multiplications to control expression blowup and thus
+            speed up calculation.
+
+        Returns
+        =======
+
+        ret : [(eigenval, multiplicity, eigenspace), ...]
+            A ragged list containing tuples of data obtained by ``eigenvals``
+            and ``nullspace``.
+
+            ``eigenspace`` is a list containing the single elementwise mod
+            squared ``eigenvector`` for each eigenvalue.
+
+            ``eigenvector`` is a vector in the form of a ``Matrix``. e.g.
+            a vector of length 3 is returned as ``Matrix([a_1, a_2, a_3])``.
+
+        Raises
+        ======
+
+        MatrixError
+            If not enough roots had got computed or if any eigenvalue has
+            multiplicity greater than 1.
+
+        NonSquareMatrixError
+            If attempted to compute eigenvectors from a non-square matrix.
+
+        NonHermitianMatrixError
+            If attempted to compute eigenvectors from a non-hermitian matrix.
+
+        See Also
+        ========
+
+        eigenvals
+        eigenvects
+        """
+
+        if not self:
+            return []
+
+        if not self.is_square:
+            raise NonSquareMatrixError()
+
+        if self.is_hermitian is False:
+            raise NonHermitianMatrixError()
+
+        evals = self.eigenvals(simplify=simplify, dotprodsimp=dotprodsimp)
+
+        if any(m > 1 for m in evals.values()):
+            raise MatrixError('Matrix has an eigenvalue with multiplicity greater than 1')
```

Comment:
I think that this needs some further work, because I believe that some can still be computed if the matrix has some eigenvalues with multplicity one, even if the matrix is degenerate.

And even if it is not going to be implemented, it would better be `NotImplementedError`, I would suggest dispatching to the naive `eigenspace` and squaring if this method is not working straightforward or the matrix is not hermitian.

Reply:
Will look into that...